### PR TITLE
feature: Allow listing of acct

### PR DESCRIPTION
**Pull request by VivienGiraud**
This PR add usage of listing accounts given an app name.

```python3
>>> keyring list MyApp
['GitHub', 'Microsoft']
```

Then you will be able to access an specific account using:
```python3
>>> keyring get MyApp Microsoft
MY_PASSWORD
```

### CODE REVIEW

**Comment by VivienGiraud** [comment]
Hi @mitya57 ,
Can you tell me if you could review this feature?
I working on an open source project that will rely on this functionality.
Have a pleasant day.

**Comment by mitya57** [comment]
1. I would name the function `list_passwords` (in plural), not `list_password`.
2. Ideally we need support in all backends, not just in the macOS one.
3. The CI is failing. In particular, you forgot to import your function in `keyring/cli.py`.

Note: I'm not the chief maintainer of Keyring, @jaraco is, so he has the final word.

File: keyring/backend.py
Comment (jaraco) [inline keyring/backend.py:104]: How about name it `list_accounts` or `list_users`, since "username" and "account" are the terms used for these fields and "passwords" aren't involved in the result.

File: README.rst
Comment (jaraco) [inline README.rst:90]: Probably this should mention "system" and not "app_name", because "system" is what's mentioned above.

Comment (jaraco) [inline README.rst:90]: We should think carefully about the "command" here. It's called "list", but "list" could mean "list users" or "list systems" or "list users and systems". I wonder if we shouldn't instead allow a generic "list" that takes a string and matches any system:password where that string is found. Let's take some time to think about other use-cases other than exact matches of a system.

**Review by jaraco — CHANGES_REQUESTED** [review]
Overall, this looks good. I have a couple of comments about naming and the design.

I'm also curious - what should this project think about the unimplemented backends? Ideally, a feature like this would be launched with support in Mac, Windows, and Linux. Would you be willing to explore an implementation that works in those backends?

Finally, this change should demand some tests. Let's make sure the test suite exercises this new expectation and functionality.

**Comment by jaraco** [comment]
This PR appears to be languishing, so I'm going to close it for now. Feel free to revive the conversation and respond to the questions and we can re-open.

**Pull request closed without merging**
